components/provisioning/org.wso2.carbon.identity.provisioning/src/test/java/org/wso2/carbon/identity/provisioning/ProvisioningUtilTest.java

```diff
@@ -0,0 +1,385 @@
+                        tenantDomain, true)).thenReturn(outBoundToCarbonClaimMappings1);
+        when(mockClaimMetadataHandler.getMappingsMapFromOtherDialectToCarbon(outboundClaimDialect, null, tenantDomain,
+                true)).thenReturn(outBoundToCarbonClaimMappings);
+        when(mockClaimMetadataHandler.getMappingsMapFromOtherDialectToCarbon(outboundClaimDialect + "1", null,
+                tenantDomain, true)).thenReturn(outBoundToCarbonClaimMappings2);
+    }
+
+    private void setUpAttributeMaps() {
+
+        attributeMap = new HashMap<>();
+
+        mappings = new ClaimMapping[6];
+        mappings[0] = ClaimMapping.build(localClaimUri, remoteClaimUri, "defaultValue", false);
+        mappings[1] = ClaimMapping.build(localClaimUri + "1", remoteClaimUri + "1", null, false);
+        mappings[2] = ClaimMapping.build(localClaimUri + "1", null, null, false);
+        mappings[3] = ClaimMapping.build(null, null, null, false);
+        mappings[4] = ClaimMapping.build(localClaimUri + "2", remoteClaimUri, null, false);
+
+        Claim claim = new Claim();
+        ClaimMapping mappingWithNullLocalClaim = new ClaimMapping();
+        mappingWithNullLocalClaim.setLocalClaim(null);
+        mappingWithNullLocalClaim.setRemoteClaim(claim);
+
+        mappings[5] = mappingWithNullLocalClaim;
+
+
+        values = new ArrayList();
+        values.add("testuser");
+        values.add("SECONDARY/testuser1");
+
+        values1 = new ArrayList();
+        values1.add("testuser2");
+
+        attributeMap.put(mappings[0], values);
+        attributeMap.put(mappings[1], new ArrayList());
+        attributeMap.put(mappings[2], values1);
+        attributeMap.put(mappings[3], values);
+        attributeMap.put(mappings[4], values);
+
+        attributeMap1 = new HashMap<>();
+        attributeMap2 = new HashMap<>();
+        attributeMap2.put(mappingWithNullLocalClaim, new ArrayList<String>());
+    }
+
+    private void setUpInboundAndOutboundClaimMaps() throws Exception {
+
+        inboundClaimMap = new HashMap();
+        inboundClaimMap1 = new HashMap();
+        inboundClaimMap2 = new HashMap();
+
+        inboundClaimMap.put(localClaimUri, "testuser");
+        inboundClaimMap.put(localClaimUri + "1", "testuser1");
+        inboundClaimMap.put(localClaimUri + "2", "testuser2");
+        inboundClaimMap.put(remoteClaimUri, "testuser3");
+        inboundClaimMap.put(remoteClaimUri + "1", "testuser4");
+
+        inboundClaimMap1.put(localClaimUri, "testuser");
+        inboundClaimMap1.put(localClaimUri + "1", "testuser1");
+
+        outboundClaimValueMappings = new HashMap();
+        outboundClaimValueMappings.put(mappings[0], values);
+        outboundClaimValueMappings.put(mappings[1], values1);
+    }
+
+    @Test(dataProvider = "getClaimValueData")
+    public void testGetClaimValues(Map<ClaimMapping, List<String>> attributeMap,
+                                   String claimUri, String userStoreDomainName) throws Exception {
+
```

Comment:
Are there any reasons for throwing generic exception here?

Reply:
This is fine. As a practice we throw generic Exception instead of specific exceptions while writing tests.